freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/upgrade/ccm/action/UpgradeCcmActions.java

```diff
@@ -183,15 +177,29 @@ protected void doExecute(UpgradeCcmContext context, UpgradeCcmEvent payload, Map
             @Override
             protected void doExecute(UpgradeCcmContext context, UpgradeCcmEvent payload, Map<Object, Object> variables) {
                 LOGGER.info("FreeIPA CCM upgrade finished {}", payload);
-                upgradeCcmService.finishedState(context.getStack().getId());
-                completeOperation(context.getStack().getAccountId(), context.getStack().getEnvironmentCrn(), variables);
-                sendEvent(context, UPGRADE_CCM_FINISHED_EVENT.event(), new StackEvent(context.getStack().getId()));
+                StackEvent stackEvent = new StackEvent(context.getStack().getId());
+                String event = UPGRADE_CCM_FINISHED_EVENT.event();
+                try {
+                    upgradeCcmService.finishedState(context.getStack().getId(), payload.getMinaRemoved());
```

Comment:
you should avoid calling salt commands in actions

Reply:
ok